### PR TITLE
headers: Return number of header removals

### PR DESCRIPTION
**Pull request by LisaLudique**
Description: This PR sets a return value for remove and removePrefix functions to be the number of headers removed as a result of the function calls rather than having them be void. The intent is to make these function calls more informative to the caller and to set a potential foundation for adding stats around when headers are removed.
Risk Level: Low, as there was no return value prior and the removal behavior itself was not changed.
Testing: Unit tests in test/common/http/header_map_impl_test.cc
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by LisaLudique** [comment]
cc @RyanMcG

**Comment by LisaLudique** [comment]
I updated the return type for the remaining header removal functions, including the one for inline headers.

File: source/common/http/header_map_impl.cc [hunk 1]
```diff
@@ -518,7 +518,8 @@ void HeaderMapImpl::clear() {
   cached_byte_size_ = 0;
 }
 
-void HeaderMapImpl::remove(const LowerCaseString& key) {
+size_t HeaderMapImpl::remove(const LowerCaseString& key) {
+  size_t old_size = headers_.size();
```
Comment (mattklein123) [inline source/common/http/header_map_impl.cc]: nit: const size_t, same elsewhere.

File: source/common/http/header_map_impl.cc [hunk 2]
```diff
@@ -534,9 +535,12 @@ void HeaderMapImpl::remove(const LowerCaseString& key) {
     }
   }
   verifyByteSize();
+  size_t new_size = headers_.size();
```
Comment (mattklein123) [inline source/common/http/header_map_impl.cc]: nit: const size_it or just kill the local variable, same elsewhere.

File: source/common/http/header_map_impl.cc [hunk 3]
```diff
@@ -611,17 +615,19 @@ HeaderMapImpl::HeaderEntryImpl* HeaderMapImpl::getExistingInline(absl::string_vi
...
   }
 
+  const size_t old_size = headers_.size();
   HeaderEntryImpl* entry = *ptr_to_entry;
   const uint64_t size_to_subtract = entry->entry_->key().size() + entry->entry_->value().size();
   subtractSize(size_to_subtract);
   *ptr_to_entry = nullptr;
   headers_.erase(entry->entry_);
   verifyByteSize();
+  return old_size - headers_.size();
```
Comment (mattklein123) [inline source/common/http/header_map_impl.cc]: Sorry I didn't mention this before, but in this case you can just return 1, right? It's guaranteed?

Comment (LisaLudique) [reply source/common/http/header_map_impl.cc]: yep, fixing

**Comment by mattklein123** [comment]
Apologies, can you merge master?

/wait

**Pull request merged**